### ISSUE TITLE
False positive for "'array_key_exists(...)' should be used instead."

### ISSUE THREAD

**Issue by sventendo** [kalessil/phpinspectionsea#1653]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.10 |
| Language level | e.g. PHP 7.4 |

Current behavior
--
This code reports the inspection "[EA] 'array_key_exists('_all', $generators)' should be used instead. It is safe to refactor for type-safe code when the indexes are integers/strings only.":
```
function foo(array $include)
{
    $generators = [
        'foo' => static function () {},
        'bar' => static function () {},
    ];

    if (in_array('_all', $include, true)) {
        $include = array_keys($generators);
    }
}
```

The suggested refactoring would change code logic.

Expected behaviour
--
No inspection in reported.

Environment details
---
PhpStorm 2020.1.4
Build #PS-201.8743.18, built on July 22, 2020

**Comment by ea-inspections-team** [comment]
Thanks for reporting @sventendo!

We looked into the possible fixes, but we are lacking the needed APIs to address this, unfortunately.
Possible workarounds are code tweaks to avoid variables names collision or suppression.
